### DISCUSSION TITLE
Documentation questions

### DISCUSSION THREAD

**Discussion by rvjansen** [adesutherland/CREXX#243] [General]
Or questions for the documentation. I decided to start work on F0029 and the Principles of Operation document, to document what is there (for someone who maybe wants to put an FPGA together, and for ourselves) and maybe work a little bit ahead on the native ABI that will come. So I am going to ask really dumb questions here, to be on the safe side.

**Comment by rvjansen** [comment]
The first one is: on 24bit VM/370, with MSB first (bit 0 on the left) is the layout of a register any different? Would that matter for external interfaces?

**Reply by adesutherland** [reply]
The way I figure it is that the actual memory pointer is in fact 32 bits. However only the first 24/31 are actually used - well sometimes the higher bits are used as status flags and the like. So CREXX uses 32 bits however when running on S/370 or whatever the higher bits will be always zero as no address will be above 24/31 bits.

**Comment by rvjansen** [comment]
Other question is, modern z/VM is still mostly 31 bit, but there is a ZCMS which is 64 bit. Are we going to support an  31 bit version for e.g. VM/ESA?

**Comment by BobBolch** [comment]
zCMS still runs in 31 bit mode.  There are some non-standard methods to allocate higher memory, but the user program has to do the mode switching.

**Reply by adesutherland** [reply]
In that case I think my answer to Rene might be no 64bit! But 31 bit on ESA for sure ...

**Reply by rvjansen** [reply]
Thanks Bob, I did not know that.

**Comment by rvjansen** [comment]
Another: where do I find the exact source of the register type/structure/union in C?

**Reply by adesutherland** [reply]
https://github.com/adesutherland/CREXX/blob/develop/interpreter/rxvmintp.h

Line 23 ...

**Comment by rvjansen** [comment]
A slight update on my activities: there is no markdown processor - know to me - that gives me acceptable typography, but with ```cmark``` I can convert wiki pages to latex. This works well and can be automated, except for some things:
1) I cannot easily add to the document index
2) some constructs, like the current table syntax, do not work. Others, like definition lists, work very well.
3) literal elements are sensitive to line length - this goes mainly for end-of-line comments

So I propose to let me edit some pages so they can be used with automatic conversions; other things I will just convert and maintain. I worked hard on it to be able to present tonight, only to realise that we had a meeting last Thursday already. So I will deliver the first version of the Principles of Operation and the updated Reference Card next week.
